network/servicenetwork/servicenetwork_test.go

```diff
@@ -343,3 +341,39 @@ func TestServiceNetwork_processIncoming(t *testing.T) {
 	_, err = serviceNetwork.processIncoming(ctx, data)
 	assert.Error(t, err)
 }
+
+func TestServiceNetwork_SetGateway(t *testing.T) {
```

Comment:
you should also check set to the same state and set to nil state behaviors

Reply:
firs case in array is nonetwork to nonetwoek